src/core/qgsabstractgeopdfexporter.cpp

```diff
@@ -338,18 +338,21 @@ QString QgsAbstractGeoPdfExporter::createCompositionXml( const QList<ComponentLa
   page.appendChild( dpi );
   // assumes DPI of 72, which is an assumption on GDALs/PDF side. It's only related to the PDF coordinate space and doesn't affect the actual output DPI!
   QDomElement width = doc.createElement( QStringLiteral( "Width" ) );
-  width.appendChild( doc.createTextNode( QString::number( std::ceil( details.pageSizeMm.width() / 25.4 * 72 ) ) ) );
+  const double pageWidthPdfUnits = std::ceil( details.pageSizeMm.width() / 25.4 * 72 );
```

Comment:
shouldn't 72 be replaced by the above DPI ? I'm a bit confused by the above comment. The DPI setting in the XML should affect the DPI of the output PDF

Reply:
Hardcoding 72 was the only way I could get things to line up - it seems the coordinate space isn't affected by the actual dpi setting itself.

Reply:
That said -- it was early in this development that I reached that conclusion, and it's possible something else was messing with things. I'll re-test.